### ISSUE TITLE
Failure to move directory to (local) external storage

### ISSUE THREAD

**Issue by skidelo** [nextcloud/server#10563]
### Steps to reproduce
1. Mount external storage of "Local" type (Call it: "local_mount")
2. Create non external storage directory at the top of the file tree (Call it: "no_mount")
3. You should now have two directories at your root. (local_mount and no_mount)
4. Move "no_mount" into "local_mount".

### Expected behaviour
"no_mount" dir moves inside "local_mount" dir

### Actual behaviour
The UI shows a spinning icon forever.
--

### Server configuration

**Operating system**: Docker (linuxserver.io)

**Web server:** nginx 1.12.2

**Database:** mysql Ver 15.1 Distrib 10.1.34

**PHP version:** 7.1.17

**Nextcloud version:** (see Nextcloud admin page) 13.0.5

**Updated from an older Nextcloud/ownCloud or fresh install:** Updated

**Where did you install Nextcloud from:** linuxserver.io docker

**Signing status:**
<details>
<summary>Signing status</summary>

```
No errors have been found.
```
</details>

**List of activated apps:**
<details>
<summary>App list</summary>

```
Enabled:
  - activity: 2.6.1
  - admin_audit: 1.3.0
  - admin_notifications: 1.0.1
  - announcementcenter: 3.2.1
  - bruteforcesettings: 1.1.0
  - calendar: 1.6.1
  - comments: 1.3.0
  - contacts: 2.1.5
  - dav: 1.4.7
  - federatedfilesharing: 1.3.1
  - federation: 1.3.0
  - files: 1.8.0
  - files_external: 1.4.1
  - files_pdfviewer: 1.2.1
  - files_sharing: 1.5.0
  - files_texteditor: 2.5.1
  - files_trashbin: 1.3.0
  - files_versions: 1.6.0
  - files_videoplayer: 1.2.0
  - firstrunwizard: 2.2.1
  - gallery: 18.0.0
  - logreader: 2.0.0
  - lookup_server_connector: 1.1.0
  - nextcloud_announcements: 1.2.0
  - notes: 2.4.0
  - notifications: 2.1.2
  - oauth2: 1.1.1
  - password_policy: 1.3.0
  - provisioning_api: 1.3.0
  - richdocuments: 2.0.10
  - serverinfo: 1.3.0
  - sharebymail: 1.3.0
  - spreed: 3.2.5
  - survey_client: 1.1.0
  - systemtags: 1.3.0
  - theming: 1.4.5
  - twofactor_backupcodes: 1.2.3
  - updatenotification: 1.3.0
  - workflowengine: 1.3.0
Disabled:
  - encryption
  - ownbackup
  - user_external
  - user_ldap
```
</details>

**Nextcloud configuration:**
<details>
<summary>Config report</summary>

```
        "memcache.local": "\\OC\\Memcache\\APCu",
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
           *** REMOVED BY REPORTER ***
        ],
        "overwriteprotocol": "https",
        "overwrite.cli.url":  *** REMOVED BY REPORTER ***
        "overwritehost":  *** REMOVED BY REPORTER ***
        "dbtype": "mysql",
        "version": "13.0.5.2",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "mail_smtpmode": "smtp",
        "mail_smtpauthtype": "LOGIN",
        "mail_smtpauth": 1,
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpsecure": "ssl",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "465",
        "maintenance": false,
        "theme": "",
        "loglevel": 2
    }
```
</details>

**Are you using external storage, if yes which one:** local

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** no

### Client configuration
**Browser:** Chrome Version 67.0.3396.99 (Official Build) (64-bit)

**Operating system:** Mac OS Sierra Version 10.12.6 (16G1212)

#### Web server error log
<details>
<summary>Web server error log</summary>

```
No known errors
```
</details>

#### Nextcloud log (data/nextcloud.log)
<details>
<summary>Nextcloud log</summary>

```
rename(): The first argument to copy() function cannot be a directory at /config/www/nextcloud/lib/private/Files/Storage/Local.php#270

rename(<no_mount>,<local_mount/no_mount>): Cross-device link at /config/www/nextcloud/lib/private/Files/Storage/Local.php#270
```
</details>

#### Browser log
<details>
<summary>Browser log</summary>

```
Insert your browser log here, this could for example include:
Console log:
client.js?v=77a6b990-0:421 Uncaught (in promise) TypeError: Cannot read property 'getElementsByTagNameNS' of null
    at Client._getSabreException (client.js?v=77a6b990-0:421)
    at client.js?v=77a6b990-0:775
```
</details>

**Comment by szaimen** [comment]
I'm closing this issue due to inactivity. If this is still happening please make sure to **upgrade** to the **latest** version. After that, feel free to reopen.

**Comment by joshtrichards** [comment]
Fixed in #38623
